### ISSUE TITLE
Upgrade from 19.0.6 to 20.0.3 fails with Doctrine\DBAL\Schema\SchemaException

### ISSUE THREAD

**Issue by hstein2m** [nextcloud/server#24652]
Hi Nextcloud community, 

I just tried to upgrade my Nextcloud setup from 19.0.6 to 20.0.3 and get the following error message in the web updater:

> Doctrine\DBAL\Schema\SchemaException: An index with name 'user_status_tstmp_ix' was already defined on table 'oc_user_status'. 

The detailed log is not much more helpful, but here it is (unfortunately in german):
> Update vorbereiten
> Log-Level auf "debug" gesetzt
> Wartungsmodus eingeschaltet
> Reparaturschritt: Repair MySQL collation
> Reparaturinformation: All tables already have the correct collation -> nothing to do
> Reparaturschritt: Repair SQLite autoincrement
> Reparaturschritt: Copy data from accounts table when migrating from ownCloud
> Reparaturschritt: Drop account terms table when migrating from ownCloud
> Das Datenbankschema wird aktualisiert
> Datenbank aktualisiert
> Suche nach einer Aktualisierung für die App "accessibility" im App-Store
> App-Store auf Aktualisierung für die App "accessibility" geprüft
[... several more Apps with the same message, probably not relevant ...]
> Suche nach einer Aktualisierung für die App "workflowengine" im App-Store
> App-Store auf Aktualisierung für die App "workflowengine" geprüft
> Doctrine\DBAL\Schema\SchemaException: An index with name 'user_status_tstmp_ix' was already defined on table 'oc_user_status'.

The server is running on Ubuntu 20.04.1, Nextcloud runs on a fairly standard Apache2 with mysql-db. 

Any ideas what I can do to fix this? 

Thank you and best regards!
Holger

**Comment by hstein2m** [comment]
Good morning, I already have an update myself: I fixed the problem by renaming the respective index. Dropping it would also have been a solution, I guess. Then, the upgrade finished without additional erros and I was able to re-create the index through an occ command. 

Nevertheless, I leave the issue open since there might be an issue with the upgrade procedure that caused this hickup in the first place. 

Best regards!

**Comment by mlsxlist** [comment]
It would be helpful if you could describe in detail what you have exactly done. That would help others if the run into the same issue.

**Comment by hstein2m** [comment]
The following steps lead to the error:
1) I updated my Nextcloud from 18.0.10 to 19.0.6 successfully via the Web Updater 
2) I started the update from 19.0.6 to 20.0.3, again via the Web Updater
3) The update failed with the message "Doctrine\DBAL\Schema\SchemaException: An index with name 'user_status_tstmp_ix' was already defined on table 'oc_user_status'." as posted before.

To fix the problem, I performed the following steps:
1) Connect to the mysql server using MySQL Workbench from my desktop computer and accessed the "nextcloud" database
2) Submit the following command: `alter table oc_user_status rename index user_status_tstmp_ix to user_status_tstmp_bak;`
3) Reload / refresh the Web Updater and start it again - this time, everything went fine and the update got installed. 
4) Log into Nextcloud with an administrator account and open Settings -> Administration -> Overview to check for installation problems.
5) Run the commands proposed there to re-create missing indexes and fix database tables: log on to the server using ssh, cd into /var/www/nextcloud (or where your nextcloud installation is residing) and run `sudo php occ [command]`.

I hope this helps other!
Best regards,
Holger
